server/services/employers.js

```diff
@@ -50,7 +50,29 @@ const updateSite = async (id, site) => {
   return dbClient.db[collections.EMPLOYER_SITES].updateDoc({ id }, changes);
 };
 
-const getSites = async () => dbClient.db[collections.EMPLOYER_SITES].findDoc({});
+/**
+ * Get all accessible sites for a user
+ * @param {*} user user with roles and sites to filter
+ * @returns list of sites which the user has access to
+ */
+
+const getSites = async (user) => {
+  const criteria = {};
+
+  if (user.isHA) {
+    criteria.siteId = user.sites.map((site) => `${site}`); // cast to string
```

Comment:
Why is the casting necessary?

Reply:
I looked into this a bit further and it looks like some MassiveJS quirkiness, I double checked the database and both of these values are stored as numbers but passing in an array of numbers to `criteria.siteId` results in this error from Massive

```
error: operator does not exist: text = integer
```

I tried [casting ](https://massivejs.org/docs/criteria-objects#casting) `siteId` to INT in the `fields` option which gets it to return a number but still fails when trying to query with a number 🙃 

Without casting: 
![image](https://user-images.githubusercontent.com/71518072/188681272-be37c5b7-7452-4c27-bdfb-d2fc0c435547.png)

![image](https://user-images.githubusercontent.com/71518072/188681340-08525ddb-dbd9-4b7c-bde4-55acd1654e26.png)


With casting:

![image](https://user-images.githubusercontent.com/71518072/188681196-06db8e62-81ca-46c0-9eff-ea5319c2b69e.png)

![image](https://user-images.githubusercontent.com/71518072/188681164-b605e80e-7d08-4617-bfc3-d778dd670636.png)